### PR TITLE
Improve terra gas price reliability

### PR DESCRIPTION
**Pull request by connorwstein**
Needs https://github.com/smartcontractkit/chainlink-terra/pull/32 merged first

### CODE REVIEW

File: core/web/assets/index.html [hunk 1]
```diff
@@ -1,3 +1 @@
-<html>
-    This is a placeholder file. Please run `make operator-ui` from the project root.
-</html>
+<!doctype html><html><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width,initial-scale=1"/><link href="https://fonts.googleapis.com/css?family=Roboto:300,400,500" rel="stylesheet"/><link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet"/><link href="/favicon.ico" rel="shortcut icon" type="image/x-icon"/><title>Chainlink</title><script defer="defer" src="/assets/main.885b9c2ad60c246d2f7a.js"></script></head><body><noscript>You need to enable JavaScript to run this app.</noscript><div id="root"></div></body></html>
```
Comment (connorwstein) [inline core/web/assets/index.html:1]: happened from just doing a make chainlink

File: core/chains/terra/terratxm/txm.go [hunk 2]
```diff
@@ -154,7 +156,7 @@ func (txm *Txm) sendMsgBatch() {
 			// after it was added for this to happen. Retry on next poll should the key be re-added.
 			continue
 		}
-		txm.sendMsgBatchFromAddress(gp, sender, key, msgs)
+		txm.sendMsgBatchFromAddress(prices["uluna"], sender, key, msgs)
```
Comment (jmank88) [inline core/chains/terra/terratxm/txm.go:159]: What happens if `prices["uluna"]` is empty/zero/unset? Would it be worth checking up front before the loop, to short circuit with one clean error?

**Comment by cl-homu** [comment]
:umbrella: The latest upstream changes (presumably #5793) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

File: core/chains/terra/terratxm/txm.go (hunk 2 above)
Comment (connorwstein) [reply core/chains/terra/terratxm/txm.go:159]: that should be impossible/would be a programmer error but couldn't hurt

File: core/chains/terra/chain.go [hunk 3]
```diff
@@ -46,12 +49,24 @@ func NewChain(db *sqlx.DB, ks keystore.Terra, logCfg pg.LogConfig, eb pg.EventBr
...
+		terraclient.NewCachingGasPriceEstimator(gpeFCD, lggr),
+		terraclient.NewClosureGasPriceEstimator(func() (map[string]sdk.DecCoin, error) {
+			return map[string]sdk.DecCoin{
+				"uluna": sdk.NewDecCoinFromDec("uluna", cfg.FallbackGasPriceULuna()),
+			}, nil
+		}),
+	}, lggr)
 	client, err := terraclient.NewClient(dbchain.ID,
-		node.TendermintURL, node.FCDURL, DefaultRequestTimeoutSeconds, lggr.Named("Client"))
+		node.TendermintURL, DefaultRequestTimeoutSeconds, lggr.Named("Client"))
```
[the remark this answers is not in the document]
Comment (connorwstein) [reply core/chains/terra/chain.go]: already addressed

File: core/chains/terra/terratxm/txm.go [hunk 4]
```diff
@@ -37,10 +37,11 @@ type Txm struct {
 	ks         keystore.Terra
 	stop, done chan struct{}
 	cfg        terra.Config
+	gpe        terraclient.ComposedGasPriceEstimator
```
[the remark this answers is not in the document]
Comment (connorwstein) [reply core/chains/terra/terratxm/txm.go:40]: the composed one doesn't actually implement the interface + it forces users to consider having a multiple estimators

**Review by jmank88 — APPROVED** [review]

**Review by krebernisak — APPROVED** [review]

**Comment by cl-homu** [comment]
:umbrella: The latest upstream changes (presumably 995bb1b73bb09f637cc9927dbd8371e2b6122cd0) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

**Review by jmank88 — APPROVED** [review]

**Pull request merged**